youtube_dl/extractor/youtube.py

```diff
@@ -906,6 +908,10 @@ class YoutubeIE(YoutubeBaseInfoExtractor):
             'url': 'https://invidio.us/watch?v=BaW_jenozKc',
             'only_matching': True,
         },
+        {
+            'url': 'https://redirect.invidious.io/watch?v=BaW_jenozKc',
```

Comment:
There is no video on this page.

Reply:
so is on the test above, which is invidio.us

please read the PR description

> Add invidious-redirect websites, i.e. websites running invidious-redirect, they redirect to all public invidious instances and then are often used to share URLs

and then (maybe) please read [invidious-redirect's README](https://github.com/iv-org/invidious-redirect)

Reply:
At the moment of writing, https://invidio.us/watch?v=BaW_jenozKc had BaW_jenozKc embedded so this patterns is kept for compatibility with old URLs possibly still floating over the net. https://redirect.invidious.io/watch?v=BaW_jenozKc does not embed any video and apparently never did thus has nothing to do in this case.

Reply:
ok then what about youtu.be? there is no video on youtu.be, it redirects to youtube.com/
same with redirect.invidious.io

even if there is no video the url path on redirect.invidious.io is right now invidious's same path and it **is** used for URLs floating over the net to share youtube videos.

Reply:
This does not matter, if you open https://youtu.be/BaW_jenozKc there is a video embedded.
If you open https://redirect.invidious.io/watch?v=BaW_jenozKc there is no video embedded.
>used for URLs floating over the net to share youtube videos.

Prove that.

Reply:
> Prove that.

some software use invidious-redirect (redirect.invidious.io) as YouTube URL rewrite:
- [bibliogram](https://git.sr.ht/~cadence/bibliogram/tree/master/item/src/lib/constants.js#L111)
- [cloudtube](https://git.sr.ht/~cadence/cloudtube/tree/main/item/pug/video.pug#L62)

there are also other software you can change YouTube rewrite URL (to another YouTube front-end or invidious-redirect) and it you can set invidious.redirect.io in config file (default, server) or preferences (user, stored with cookies)
- [nitter](https://github.com/zedeus/nitter)

some websites use invidious-redirect (redirect.invidious.io) as YouTube URL rewrite:
- [nitter.pussthecat.org](https://github.com/PussTheCat-org/PussTheCat.org-Configs/blob/master/Services/Nitter/nitter.conf#L26)

if you want **actual content**:
- https://nitter.pussthecat.org/YouTube
- https://nitter.pussthecat.org/YouTube/status/1360363141947944964#m
- https://tube.cadence.moe/watch?v=BaW_jenozKc (look for the invidious link)
- …